### ISSUE TITLE
Very slow to create joint normal random symbol for covariance matrix with symbolic elements

### ISSUE THREAD

**Issue by rikardn** [sympy/sympy#18625]
If I create a symmetric matrix with symbolic elements and use that as covariance matrix for a joint normal distribution the program seemingly freezes. Aborting the program I see that it is stuck in the ```is_positive_definite``` method. The code below is an example of this happening. Using a size of 3 or 2 will work and give a result quickly.

I guess in this situation the positive definiteness is unknown. Could perhaps the positive definiteness check give true or unknown in these cases so that the random variable could be created?

```python
from sympy import Matrix, symarray, stats, pprint

size = 4
symbs = symarray('a', (size, size), real=True)

# Make matrix symmetric
for i in range(len(symbs)):
    for j in range(i, len(symbs)):
        if i != j:
            symbs[i, j] = symbs[j, i]

A = Matrix(symbs)
pprint(A)

X = stats.Normal('X', [0] * size, A)
```

After aborting:

```
Traceback (most recent call last):

  File "<ipython-input-51-c4b023a53800>", line 14, in <module>
    X = stats.Normal('X', [0] * size, A)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/crv_types.py", line 2658, in Normal
    MultivariateNormalDistribution, name, mean, std)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv_types.py", line 27, in multivariate_rv
    dist.check(*args)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/stats/joint_rv_types.py", line 92, in check
    _value_check(sigma.is_positive_definite,

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/matrices/matrices.py", line 1533, in is_positive_definite
    return self._eval_is_positive_definite()

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/matrices/matrices.py", line 1507, in _eval_is_positive_definite
    eigen = self.eigenvals()

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/matrices/matrices.py", line 1278, in eigenvals
    eigs = roots(mat.charpoly(x=Dummy('x')), **flags)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyroots.py", line 1017, in roots
    for currentroot in _try_decompose(f):

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyroots.py", line 915, in _try_decompose
    for currentroot in _try_heuristics(factors[0]):

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyroots.py", line 961, in _try_heuristics
    result += roots_quartic(f)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyroots.py", line 322, in roots_quartic
    sols = _roots_quartic_euler(e, f, g, aon4)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyroots.py", line 228, in _roots_quartic_euler
    xsols = list(roots(Poly(eq, x), cubics=False).keys())

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyroots.py", line 996, in roots
    _, factors = Poly(f.as_expr()).factor_list()

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polytools.py", line 3240, in factor_list
    coeff, factors = f.rep.factor_list()

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/polyclasses.py", line 796, in factor_list
    coeff, factors = dmp_factor_list(f.rep, f.lev, f.dom)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/factortools.py", line 1289, in dmp_factor_list
    coeff, factors = dmp_zz_factor(f, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/factortools.py", line 1099, in dmp_zz_factor
    g = dmp_sqf_part(g, u, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/sqfreetools.py", line 255, in dmp_sqf_part
    gcd = dmp_gcd(f, dmp_diff(f, 1, u, K), u, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1627, in dmp_gcd
    return dmp_inner_gcd(f, g, u, K)[0]

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1586, in dmp_inner_gcd
    h, cff, cfg = _dmp_inner_gcd(f, g, u, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1555, in _dmp_inner_gcd
    return dmp_zz_heu_gcd(f, g, u, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1336, in dmp_zz_heu_gcd
    h, cff, cfg = dmp_zz_heu_gcd(ff, gg, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/euclidtools.py", line 1330, in dmp_zz_heu_gcd
    ff = dmp_eval(f, x, u, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/densetools.py", line 313, in dmp_eval
    result = dmp_mul_ground(result, a, v, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/densearith.py", line 292, in dmp_mul_ground
    return dup_mul_ground(f, c, K)

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/densearith.py", line 274, in dup_mul_ground
    return [ cf * c for cf in f ]

  File "/home/rikard/.local/lib/python3.7/site-packages/sympy/polys/densearith.py", line 274, in <listcomp>
    return [ cf * c for cf in f ]

KeyboardInterrupt
```

**Comment by czgdp1807** [comment]
The following hangs,
https://github.com/sympy/sympy/blob/5b92c4497fcc6f1df4aac23b9c001ff323ffb421/sympy/matrices/eigen.py#L600
Try out,
```python
print(A.eigenvals())
```

**Comment by Smit-create** [comment]
I think the above issue with the matrices has been fixed like I get the following results:
```py
In [1]: from sympy import Matrix, symarray, stats, pprint

In [2]: size = 4

In [3]: symbs = symarray('a', (size, size), real=True)

In [4]: for i in range(len(symbs)):
   ...:     for j in range(i, len(symbs)):

   ...:         if i != j:
   ...:             symbs[i, j] = symbs[j, i]
   ...:

In [5]: A = Matrix(symbs)

In [6]: pprint(A)
⎡a₀ ₀  a₁ ₀  a₂ ₀  a₃ ₀⎤
⎢                      ⎥
⎢a₁ ₀  a₁ ₁  a₂ ₁  a₃ ₁⎥
⎢                      ⎥
⎢a₂ ₀  a₂ ₁  a₂ ₂  a₃ ₂⎥
⎢                      ⎥
⎣a₃ ₀  a₃ ₁  a₃ ₂  a₃ ₃⎦

In [7]: X = stats.Normal('X', [0] * size, A)

In [8]: X
Out[8]: X

In [9]: %timeit X = stats.Normal('X', [0] * size, A)
1.76 ms ± 7.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

```
For size=3:
```py
In [15]: %timeit X = stats.Normal('X', [0] * size, A)
1.16 ms ± 17.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

```
For size=2:
```py
In [22]: %timeit X = stats.Normal('X', [0] * size, A)
746 µs ± 17.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Does this issue looks fixed with matrices @oscarbenjamin?

**Comment by oscarbenjamin** [comment]
I think this was fixed in #19205

**Comment by rikardn** [comment]
I can confirm that this issue no longer exists in sympy 1.6.1.

I tested  ```size = 20``` in my example and it was without perceivable slowdown. 20x20 would be more than what my applications will need.

Thanks to all involved in fixing it!
